eZ/Publish/Core/Persistence/Legacy/Tests/Content/StorageHandlerTest.php

```diff
@@ -127,6 +128,34 @@ public function testGetFieldDataNotAvailable()
             ->with($this->equalTo('foobar'))
             ->will($this->returnValue($storageMock));
 
+        $field = new Field();
+        $field->id = 123;
+        $field->type = 'foobar';
+        $field->value = new FieldValue();
+
+        $handler = $this->getStorageHandler();
+        $handler->getFieldData($this->getVersionInfoMock(), $field);
+    }
+
+    /**
+     * @covers \eZ\Publish\Core\Persistence\Legacy\Content\StorageHandler::getFieldData
+     */
+    public function testGetFieldDataNotAvailableForVirtualField()
+    {
+        $storageMock = $this->getStorageMock();
+        $storageRegistryMock = $this->getStorageRegistryMock();
+
+        $storageMock->expects(self::never())
+            ->method('hasFieldData');
+
+        $storageMock->expects(self::never())
+            ->method('getFieldData');
+
+        $storageRegistryMock->expects(self::once())
+            ->method('getStorage')
+            ->with($this->equalTo('foobar'))
```

Comment:
```suggestion
            ->with(self::equalTo('foobar'))
```
or (since it's exactly the same as)
```suggestion
            ->with('foobar')
```

You can also use
```suggestion
            ->with(self::identicalTo('foobar'))
```
which is as if you used `assertSame` assertion (with strict comparison and all that).